### DISCUSSION TITLE
Using `Feature.final_status` (or better not)

### DISCUSSION THREAD

**Discussion by Marcin-Mach-Samsung** [behave/behave#1198] [Q&A] [answered]
Hi.
I have simple testing `tutorial.feature` file:

```gherkin
Feature: showing off behave

  Scenario: fun a simple test
    Given we have behave installed
    When we implement a test
    Then behave will test it for us!

  @wip
  Scenario Outline: Use blender with <thing>
    Given I put "<thing>" in a blender,
    When I switch the blender on
    Then "<kind>" should transform into "<other thing>"


    Examples: Amphibians
      | thing         | other thing |  kind   |
      | Red Tree Frog | mush        |  animal |
      | apples        | apple juice |  fruit  |
```

Nothing special... I've also environment.py with below **after_feature** definition:

```py
def after_feature(context, feature):
    f = open("mlog.txt", "a")
    f.write(f"feature status: {feature.status} err msg: {feature.error_message} ")
    for fs in feature.final_status:
        f.write(f"{fs.name}: {fs.value}\n")
    f.close()
```

Sounds simple... now, when I'm running `behave --tags=wip` which should skip the first scenario and execute only the second one (outline) two times, I'm getting summary as expected:

```
1 feature passed, 0 failed, 0 skipped
2 scenarios passed, 0 failed, 1 skipped
6 steps passed, 0 failed, 3 skipped, 0 undefined
Took 0m0.000s
```

but in the `mlog.txt` file I'm getting something surprised me:

```
after tag: wip
feature status: Status.passed err msg: None
passed: 2
failed: 3
skipped: 1
```

The question is: 
* why the failed is equal to 3. 
* How did behave calculate it. 
* I expected rather equal to '0'.

**Comment by Marcin-Mach-Samsung** [comment]
Ok... nvm. I found it's my misunderstanding of this field. This is not the counter but enum of possible statuses. I'm right?

**Reply by jenisys** [reply, marked as answer]
Not quite: 
* `final_status` is a constant class-attribute of a Feature base-class
* `final_status` is basically a list of `Status` enum-values that indicate that the model-element (in this case the `Feature`) has reached a final state. Therefore, it is the subset of `Status` enum-values that have the `final` „property“ (and should not change afterwards in the test run).

**Comment by jenisys** [comment]
@Marcin-Mach-Samsung 
What you are probably trying to achieve in the part with `feature.final_status` is to get an overview of the other outcomes of this feature. This is basically the `summary` functionality that is printed at the end by your test-runner.

SEE ALSO:

* [behave.reporter.summary](https://github.com/behave/behave/blob/main/behave/reporter/summary.py) module and its `SummaryReporter` class

**Comment by jenisys** [comment]
@Marcin-Mach-Samsung 
HINTS:
* `final_status` was removed from the model-element base class to remove such confusions in the future (replaced by: `Status.is_final()`)
* The new [behave.summary](https://github.com/behave/behave/blob/main/behave/summary.py) module and its classes may help you to determine the summary counts (if this was your intention).
